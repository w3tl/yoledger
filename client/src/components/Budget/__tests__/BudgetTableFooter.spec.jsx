import React from 'react';
import { mount } from 'enzyme';
import { Table } from 'semantic-ui-react';
import RawBudgetTableFooter from '../BudgetTableFooter';

const BudgetTableFooter = props => (
  <Table>
    <RawBudgetTableFooter
      accounts={[]}
      {...props}
    />
  </Table>
);

describe('BudgetTableFooter', () => {
  it('should call onCreate add button was clicked', () => {
    const onCreate = jest.fn();
    const output = mount(
      <BudgetTableFooter onCreate={onCreate} initialState={{ isAddAccount: true, newAccount: 'New account' }} />,
    );
    output.find('Button[id="addAccount"]').simulate('click');
    expect(onCreate.mock.calls[0]).toMatchSnapshot('');
  });
});
