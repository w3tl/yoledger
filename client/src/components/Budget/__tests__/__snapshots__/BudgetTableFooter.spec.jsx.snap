// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BudgetTableFooter should call onCreate add button was clicked 1`] = `
Array [
  Object {
    "account": "New account",
  },
]
`;
